pkg/reporting/reporting.go

```diff
@@ -55,20 +55,41 @@ func updateReport(e *ExtendedVulnerability, c *config.Check) {
 
 func parseReports(reports map[string]*report.Report, cfg *config.Config, l log.Logger) []ExtendedVulnerability {
 	vulns := []ExtendedVulnerability{}
-	for _, r := range reports {
+
+	for _, check := range cfg.Checks {
+		// The check was filtered
+		if check.Id == "" {
+			continue
+		}
+
+		// See if the check received a report
+		r, ok := reports[check.Id]
+
+		// Write a log in case of failure and a missing req variable
+		if !ok || r.Status != "FINISHED" {
+			lv := []string{}
+			for _, requiredVar := range check.Checktype.RequiredVars {
+				if val, ok := cfg.Conf.Vars[requiredVar]; !ok || len(val) == 0 {
+					lv = append(lv, requiredVar)
+				}
+			}
+			if len(lv) > 0 {
+				l.Errorf("Check %s on %s failed and variables where missing %v", check.Checktype.Name, check.Target, lv)
+			}
+		}
```

Comment:
I think this function has grown a lot. A symptom of it is that in order to test this new part you had to set up a test with a lot of input data. It might be a good idea to extract this part to its own function. Although it doesn't look like it could be reusable and is well documented, visually, for my taste it looks a bit like spaghetti code. WDYT?

Reply:
Good catch, the validation can easily moved out. I pushed a refactor [6cce6ed](https://github.com/adevinta/vulcan-local/pull/36/commits/6cce6eddf868827629ab20794fd82355cdb46472)